orbstation/antagonists/wizard_journeyman/grand_ritual/grand_ritual_finale.dm

```diff
@@ -0,0 +1,380 @@
+/datum/grand_finale/midas/trigger(mob/living/carbon/human/invoker)
+	var/theme_path = pick(permitted_transforms)
+	chosen_theme = new theme_path()
+	var/turf/start_turf = get_turf(invoker)
+	var/greatest_dist = 0
+	var/list/turfs_to_transform = list()
+	for (var/turf/transform_turf as anything in GLOB.station_turfs)
+		if (!chosen_theme.can_convert(transform_turf))
+			continue
+		var/dist = get_dist(start_turf, transform_turf)
+		if (dist > greatest_dist)
+			greatest_dist = dist
+		if (!turfs_to_transform["[dist]"])
+			turfs_to_transform["[dist]"] = list()
+		turfs_to_transform["[dist]"] += transform_turf
+
+	if (chosen_theme.can_convert(start_turf))
+		chosen_theme.apply_theme(start_turf)
+
+	for (var/iterator in 1 to greatest_dist)
+		if(!turfs_to_transform["[iterator]"])
+			continue
+		addtimer(CALLBACK(src, .proc/transform_area, turfs_to_transform["[iterator]"]), (5 SECONDS) * iterator)
+
+/datum/grand_finale/midas/proc/transform_area(list/turfs)
+	for (var/turf/transform_turf as anything in turfs)
+		if (!chosen_theme.can_convert(transform_turf))
+			continue
+		chosen_theme.apply_theme(transform_turf)
+
+/// Kill yourself and probably a bunch of other people
+/datum/grand_finale/armageddon
+	name = "Annihilation"
+	desc = "This crew have offended you beyond the realm of pranks. Make the ultimate sacrifice to teach them a lesson your elders can really respect. \
+		YOU WILL NOT SURVIVE THIS."
+	icon = 'icons/hud/screen_alert.dmi'
+	icon_state = "wounded"
+	minimum_time = 90 MINUTES // This will probably immediately end the round if it gets finished.
+	ritual_invoke_time = 60 SECONDS // Really give the crew some time to interfere with this one.
+	dire_warning = TRUE
+	glow_colour = "#be000048"
+	/// Things to yell before you die
+	var/static/list/possible_last_words = list(
+		"Flames and ruin!",
+		"Dooooooooom!!",
+		"The hearts of men are black with corruption and must needs be cleansed!",
+		"Death, and death alone!",
+		"Ruination is come!",
+		"Raise your swords against the coming night!",
+		"Your journey ends here!",
+		"The gods will not be watching.",
+		"There is time enough for regret in the flames of hell.",
+		"And now the scales will tip!",
+		"Even the strongest of shields cannot defend the weakest of wills!",
+		"You shall rue the day you raised your eyes to the heavens.",
+		"Denizens of the abyss! From ink of blackest night, I summon you! Darkness to me!",
+		"Your very soul shall not escape my wrath!", )
+
+/datum/grand_finale/armageddon/trigger(mob/living/carbon/human/invoker)
+	priority_announce(pick(possible_last_words), null, 'sound/magic/voidblink.ogg', sender_override = "[invoker.real_name]")
+	var/turf/current_location = get_turf(invoker)
+	invoker.gib()
+	var/doom = rand(1, 4)
+	switch(doom)
+		if (1)
+			var/obj/singularity/singulo = new(current_location)
+			singulo.energy = 300
+		if (2)
+			var/obj/energy_ball/tesla = new (current_location)
+			tesla.energy = 200
+		if (3)
```

Comment:
This one concerns me a little, with regards to the likelihood of people being upset about an automatic round-end. Also, possibly takes the wind out of the blood cult's schtick a bit if a dickhead wizard can summon Nar'Sie alone, but then again that's also funny.

As the comments say, there's a whole three minutes to interrupt this, so it's exceedingly unlikely to happen anyway.

Reply:
I am not too worried about stealing the thunder of Blood Cult here because I simply don't think we will ever have a blood cult.